proposals/0296-increase-tx-size.md

```diff
@@ -0,0 +1,177 @@
+and the new v1 transaction format required to support the larger size. v1 
+transactions are a direct upgrade from the legacy format, while proposing 
+a removal of Address Lookup Tables compared to v0 transactions.
+
+## Motivation
+
+The current transaction size limit of 1232 bytes is too small for many
+developer use cases. While there have been some attempts to artificially
+increase the size capacity with address lookup tables, the current cap still is
+artificially limiting the number of signatures and size of other data.
+
+Use cases that are currently limited by the transaction size include:
+
+- ZK proofs such as what is used within [Confidential
+ Balances](https://www.solana-program.com/docs/confidential-balances/zkps)
+- Untruncated Winternitz one-time signatures
+- Nested Multisig used by corporations under Squads
+- Onchain cryptography signature schemes without precompiles such as BLS
+
+and many more. A larger transaction size would also help remove the need for
+address lookup tables, which have been a source of complexity on the validator
+client as well as a frictional developer experience for many developers.
+
+A number of developers have also resorted to using Jito bundles to artificially
+increase the size of their transactions. This is a current workaround that
+developers are using, but it does not have the same guarantees as a transaction
+does on the protocol level, namely that bundles are not guaranteed to be atomic
+like singular transactions.
+
+For these reasons, increasing the current transaction size limit would enable
+developers to create new applications and use cases that were previously
+unfeasible.
+
+## New Terminology
+
+- 'v1 transaction' - A new transaction format that is designed to enable larger
+  transactions sizes while not having the address lookup table features 
+  introduced in v0 transactions.
+
+## Detailed Design
+
+Increasing the transaction size limit above the current MTU max size is
+possible with the use of QUIC. QUIC's RFC 9000 specification does not have an
+explicit maximum stream size, allowing for larger transactions to be sent.
+
+A new transaction format, `v1`, is proposed to enable larger transaction sizes.
+The `v1` transaction format would be:
+
+```
+VersionByte (u8) - >127 to distinguish from legacy/v0 formats
+LegacyHeader (u8, u8, u8) -- Required signatures from the current
+`MessageHeader` type
+Payload Length (u16) -- Total size excluding signatures
+NumInstructions (u16)
+ResourceRequestMask (u16) -- Bitmask of which resource requests are present.
+ For example bit 0 may mean "requested_cus", bit 1 may mean "microlamports per
+ cu"
+NumStaticKeys (u8)
+RecentBlockhash (hash)
+StaticKeys [[u8; 32]] -- Number matches NumStaticKeys
+ResourceRequests [u64] -- Array of request values. (section size is popcount
+ ResourceRequestMask * 8). each value is a u64.
+Ixs [(u16, u16, u16, u16)] -- Number matches NumInstructions. Values are
```

Comment:
Why is `num_accounts` a u16 if the upper limit of ‘static keys’ above is u8? Without knowing what the format of an address is in `TRAILING_DATA_SECTION` it's hard to comment.

Reply:
It's a compressed u16 now in the serialization format. Just went with a u16 here to avoid having to change it.

Accounts can be repeated. I'm in favor of just changing it to a `u8`, since we need to also add the program_index in for each of `Ixs`:

@jacobcreech We should do:

```
Ixs [(u8, u8, u16, u16, u16)] -- Number matches NumInstructions. Values are
 (program_account_index, num_accounts, accounts_offset, num_data_bytes, bytes_offset)
```